File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](const vector<int>& a, const vector<int>& b) {
         return a[0] < b[0];
        });
        int count=0;
        int n=intervals.size();
        for(int i=1;i<n;i++){
            if(intervals[i][0]<intervals[i-1][1]){
                count++;
                intervals[i][1]=min(intervals[i][1],intervals[i-1][1]);
            }
        }
        return count ;
    }
};

File: Data Structures & Algorithms/assign-cookies/submission-2.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        if not s or not g:
            return 0
        s.sort()
        g.sort()
        n,m=len(g),len(s)
        i,j=n-1,m-1
        count=0
        while i>=0 and j>=0:
            if g[i]<=s[j]:
                i-=1
                j-=1
                count+=1
            else:
                i-=1
        return count